sos/report/plugins/microshift.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins.openshift import Openshift
+
+
+class Microshift(Openshift):
+    """
+    For Microshift users, this plugin does not need to perform any oauth steps
+    of any kind. Users should see the `microshift` plugin enable and NOT the
+    `openshift` plugin during a normal execution of `sos report`. All plugin
+    options are otherwise the same, however users should note that the `token`
+    option is ignored by the Microshift plugin variant.
+    """
+
+    short_desc = 'Microshift Container Platform'
+    plugin_name = 'microshift'
+    packages = ('microshift',)
+    services = ('microshift',)
+    profiles = ()
+    master_localhost_kubeconfig = ''  # TODO: is there a default kubeconfig?
+
+    def _check_oc_logged_in(self):
```

Comment:
This function could verify that the required kubeconfig file exists.

Reply:
The parent class does so - it's probably obscured by the diff output, but we check to see if the kubeconfig file exists, and if it does we run all our `oc` commands with the `KUBECONFIG` env var set to that location.

This function override is to simply bypass the `oc login` logic. If `oc whoami` won't work under microshift, we'd need a separate method override to run some `oc` command to verify that we're able to do so successfully (so _what_ command that is doesn't actually matter as much).

Reply:
Oh, sorry - misread your note. Ack, that should be done here.

Reply:
Realized I had the wrong method overridden here for what I was intending. The path verification is indeed done in the parent class, before calling `_check_oc_function()` which is what I had meant to originally override.

In `openshift` that is where we do `oc whoami`, and if that fails we try to do the `oc login`. As a placeholder, I'm just checking that we have `oc` available to us - if there's an actual `oc` command we can safely run for microshift then let's use that instead.

Reply:
MicroShift has a kubeadmin kubconfig on disk already, and there's no login. So we could test that the credentials work for listing something simple without an error (maybe like listing namespaces?) but there's no separate login step.

Reply:
I left a comment in https://github.com/openshift/enhancements/pull/1318/files regarding kube config file location and use of -k openshift.kubeconfig=/home/microshift/.kube/config. I'm not sure if that would be the default or the /var/lib/microshift/resources/kubeadmin/kubeconfig file.